dialect/sqlitedialect/dialect.go

```diff
@@ -91,6 +92,25 @@ func (d *Dialect) DefaultVarcharLen() int {
 	return 0
 }
 
+// AppendSequence adds AUTOINCREMENT keyword to the column definition. As per [documentation],
+// AUTOINCREMENT is only valid for INTEGER PRIMARY KEY, and this method will be a noop for other columns.
+//
+// Because this is a valid construct:
+//	CREATE TABLE ("id" INTEGER PRIMARY KEY AUTOINCREMENT);
+// and this is not:
+//	CREATE TABLE ("id" INTEGER AUTOINCREMENT, PRIMARY KEY ("id"));
+// AppendSequence adds a primary key constraint as a *side-effect*. Callers should expect it to avoid building invalid SQL.
+// SQLite also [does not support] AUTOINCREMENT column in composite primary keys.
+//
+// [documentation]: https://www.sqlite.org/autoinc.html
+// [does not support]: https://stackoverflow.com/a/6793274/14726116
+func (d *Dialect) AppendSequence(b []byte, table *schema.Table, field *schema.Field) []byte {
+	if field.IsPK && len(table.PKs) == 1 && field.CreateTableSQLType == sqltype.Integer {
+		b = append(b, " PRIMARY KEY AUTOINCREMENT"...)
+	}
+	return b
```

Comment:
Could you refactor this method to always append `PRIMARY KEY AUTOINCREMENT` and move the logic upwards to `query_table_create.go`?

Reply:
Have I understood your suggestion correctly?  

```go
// query_table_create.go

if (field.Identity && fmter.HasFeature(feature.GeneratedIdentity)) ||
	(field.AutoIncrement && (fmter.HasFeature(feature.AutoIncrement) || fmter.HasFeature(feature.Identity))) {
		
	if q.db.dialect.Name() == dialect.SQLite {
		if field.IsPK && len(q.table.PKs) == 1 && field.CreateTableSQLType == sqltype.Integer {
			b = q.db.dialect.AppendSequence(b, q.table, field)
		}
	} else {
		b = q.db.dialect.AppendSequence(b, q.table, field)
	}
}
```

If I may share some thoughts about the code above:

1. Surfacing that `if` condition in the `CreateTable` makes it easier to see when the method will actually do something.  
1. The limitations of `AUTOINCREMENT` keyword (SQLite-specific) seem like something that'd  be best described  in the dialect code.
1. Other callers of `AppendSequence()` (if there are any)  would need to know about it and do the check themselves.


WDYT?

Reply:
>Other callers of AppendSequence() (if there are any) would need to know about it and do the check themselves.

Well, the details are already leaking to other places, e.g. you have to craft `bytes.Contains` check to account for what is done in `AppendSequence`.

I would probably just remove `AppendSequence` since the abstraction it provides is not that useful and leaks to other places.

We can do it later though.

Reply:
>  the abstraction it provides is not that useful

The use case I had in mind was the ability to add dialect-specific details, like switching between `GENERATED { BY DEFAULT | IDENTITY} AS SEQUENCE` in Postgres (see #688) or setting [`seed` and `increment`](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql-identity-property?view=sql-server-ver16#syntax) in MSSQL.

> the details are already leaking to other places, e.g. you have to craft bytes.Contains

You're right, I haven't thought about that before.  
Do you think it'd be an improvement if instead `AppendSequence()` returned a `bool` indicating whether "a PRIMARY KEY can still be defined in this query"? With that being clearly documented in the `Dialect` interface.

We'll need to check if we can still `q.appendPKConstraint()` after handling the sequence either way.